### ISSUE TITLE
Render Order Issue with Stairs

### ISSUE THREAD

**Issue by NiteKat** [diasurgical/devilutionX#3450]
**Important information**
Windows 10 / b8affd5f / Built myself

**Describe**
Walking with a sword and shield equipped as warrior behind a cathedral down stairwell causes the sword to poke through the stairwell.

**To Reproduce**
1. Go to any cathedral level as a warrior with a sword and shield equipped.
2. Walk along the back of the stairs, hugging the stairs.
3. Observe the sword that can cut through stone.

**Expected behavior**
Stairs should render on top of the sword.

**Screenshots**
![swordthroughstairs](https://user-images.githubusercontent.com/28605404/140847913-794df445-508e-432d-a020-732e0d1c2f8f.gif)

**Additional context**
This is present in Vanilla 1.09 Diablo.

**Comment by julealgon** [comment]
I don't think this is exclusive to stairs, though. IIRC, the same thing happens with normal walls in that same configuration.

There was a discussion about it when trying out various ways of showing the player behind walls.

**Comment by NiteKat** [comment]
Entirely possible. I never really noticed this before, and I have stair trapped the butcher a lot. But then Ajenbo pointed it out in chat. 😆

**Comment by agris-codes** [comment]
This exists with catacombs stairs as well, the “up” ones iirc. On DX, not sure on vanilla.
